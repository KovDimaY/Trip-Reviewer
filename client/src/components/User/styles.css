.user-container {
    max-width: 450px;
    margin: 75px auto;
}
.user-container .data-wrapper {
    display: flex;
    flex-direction: row;
    margin-bottom: 50px;
}
.user-container .avatar {
    width: 200px;
    height: 200px;
}
.user-container .info-wrapper {
    margin-left: 50px;
}
.user-container .info-item {
    overflow: hidden;
    text-overflow: ellipsis;
    font-size: 18px;
    font-weight: 300;
    margin-bottom: 20px;
    color: #a4a4a4;
}
.user-container .info-title {
    display: block;
    font-weight: 500;
    font-size: 24px;
    color: #02bcd4;
    margin-bottom: 5px;
}
.user-container .button-link {
    width: 172px;
    display: inline-block;
    font-size: 20px;
    padding: 10px 15px;
    background: #02BCD4;
    color: #fff;
    font-weight: 300;
    border-radius: 3px;
    border: none;
}
.user-container .button-link:hover {
    font-size: 22px;
    background: #009EB3;
}
.user-container .button-link:active {
    font-size: 21px;
    background: #0495A8;
}

@media (max-width: 800px) {
    .user-container {
        margin: 50px auto 75px;
    }
    .user-container .data-wrapper {
        flex-direction: column;
        align-items: center;
        text-align: center;
        margin: 0;
    }
    .user-container .info-wrapper {
        margin: 50px 0;
    }
    .user-container .data .info-item {
        text-align: center;
    }
}