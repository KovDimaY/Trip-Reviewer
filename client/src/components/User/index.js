import React from 'react';
import { Link } from 'react-router-dom';

import UserAvatar from './../UserAvatar';

import './styles.css';

const User = (props) => {
    const user = props.users.login;
    const redirect = `/user/edit-profile/${user.id}`;

    return (
        <div className="user_container">
            <UserAvatar filename={user.avatar} />
            <div className="data">
                <div><span>Name:</span> {user.name}</div>
                <div><span>Lastname:</span> {user.lastname}</div>
                <div><span>Email:</span> {user.email}</div>
            </div>
            <div className="text-center">
                <Link to={redirect} className="button-link">Edit Profile</Link>
            </div>
        </div>
    );
};

export default User;