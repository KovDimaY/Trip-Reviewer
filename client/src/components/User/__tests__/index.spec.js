import React from 'react';
import { create } from 'react-test-renderer';
 
import User from './../../User';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

const mockComponent = props => {
    return (
        <User {...props} />
    );
};
 
describe('<User />', () => {
    it('should render component', () => {
        const props = {
            users: {
                login: {
                    name: 'name',
                    lastname: 'lastname',
                    email: 'email'
                }
            }
        };
        const tree = create(mockComponent(props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });
});