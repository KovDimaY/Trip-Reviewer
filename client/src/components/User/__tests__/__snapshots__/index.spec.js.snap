// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should render component 1`] = `
<div
  className="user_container"
>
  <div
    className="avatar"
  >
    <img
      alt="avatar"
      src="/images/avatar.png"
    />
  </div>
  <div
    className="nfo"
  >
    <div>
      <span>
        Name:
      </span>
       
      name
    </div>
    <div>
      <span>
        Lastname:
      </span>
       
      lastname
    </div>
    <div>
      <span>
        Email:
      </span>
       
      email
    </div>
    <div>
      <span>
        Password:
      </span>
       * * * * * * 
    </div>
  </div>
  <div
    className="text-center"
  >
    <Link
      className="button-link"
      to="/user/edit-profile/undefined"
    >
      Edit Profile
    </Link>
  </div>
</div>
`;
